.github/workflows/ci.yml

```diff
@@ -14,47 +11,58 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  go_version: '~1.20.12'
+  go_version: "~1.23.1"
+
+permissions:
+  # Required: allow read access to the content for analysis.
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  pull-requests: read
+  # Optional: allow write access to checks to allow the action to annotate code in the PR.
+  checks: write
 
 jobs:
   unit:
     name: Unit Tests
     runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Update dependencies
         run: git submodule update --init
-      - uses: actions/setup-go@v3
+      - uses: actions/setup-go@v5
         with:
           go-version: ${{ env.go_version }}
       - name: Install libolm
         run: sudo apt update && sudo apt-get install -y libolm-dev
       - name: build_test
         shell: bash
         run: ./scripts/build_test.sh
+  # When the golangci-lint version is updated here, also update it in scripts/lint.sh
   lint:
     name: Static Analysis
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```

Comment:
we agreed to use 22.04 so that we don't have different dependencies when the latest version is updated

Reply:
Fixed